### PR TITLE
Fixing MIRI wavelengths, fixing 2D LC x-axis, generalizing S6, and other patches

### LINKED ISSUE #399: [Bug]: NaN values in wave_1d array crashing Stage 3 plotting
### FAQ check

- [X] Yes, I checked the FAQ and my question has not been addressed.

### Instrument

NIRSpec (Stages 1-3)

### What happened?

NaN values are appearing in the low wavelength end of the wave_1d array in the plots_s3.lc_nodriftcorr() function (see example below). Right now I have a workaround of just not including those columns in the stage 3 by restricting the xwindow. Need to dig into where these NaN values are coming from, and if they are unavoidable, whether to replace them with values that the plotting routine can digest. 

```
<xarray.DataArray 'wave_1d' (x: 490)>
array([       nan,        nan,        nan,        nan,        nan,
              nan,        nan,        nan,        nan, 0.5021705 ,
       0.5047077 , 0.5073139 , 0.5099276 , 0.5126182 , 0.5153201 ,
       0.5180906 , 0.52088755, 0.52373844, 0.5266363 , 0.529569  ,
       0.5325738 , 0.53560734, 0.5387073 , 0.5418587 , 0.54504687,
       0.54831946, 0.55163634, 0.55499846, 0.5584488 , 0.5619505 ,
       0.5655058 , 0.56913906, 0.57284486, 0.5766127 , 0.5804462 ,
       0.5843633 , 0.588363  , 0.59243673, 0.59658813, 0.6003529 ,
       0.6046618 , 0.609068  , 0.61356777, 0.6181604 , 0.6228493 ,
       0.6276381 , 0.63253033, 0.63752943, 0.64263886, 0.64786226,
       0.65320283, 0.6586641 , 0.66424936, 0.669975  , 0.6758352 ,
       0.6818305 , 0.687965  , 0.6942429 , 0.7006931 , 0.70729166,
       0.71404576, 0.72098774, 0.7280914 , 0.7353828 , 0.7428576 ,
       0.7505296 , 0.75839496, 0.7664771 , 0.7747623 , 0.78327125,
       0.79201263, 0.80098367, 0.8101939 , 0.8196522 , 0.82936716,
       0.8393447 , 0.8495919 , 0.8601169 , 0.8709254 , 0.8820227 ,
       0.89341253, 0.90511304, 0.9171213 , 0.9294393 , 0.942083  ,
       0.95504546, 0.96833354, 0.98195785, 0.9959099 , 1.0101937 ,
       1.02481   , 1.039758  , 1.0550355 , 1.0706391 , 1.0865657 ,
       1.1028067 , 1.1193589 , 1.1362109 , 1.1533505 , 1.170776  ,
...
       4.8968735 , 4.9048734 , 4.9128594 , 4.9208307 , 4.9287887 ,
       4.9367313 , 4.9446607 , 4.952576  , 4.9604774 , 4.9683657 ,
       4.9762387 , 4.9841    , 4.9919453 , 4.9997787 , 5.007598  ,
       5.0154037 , 5.0231967 , 5.030976  , 5.038742  , 5.0464945 ,
       5.0542345 , 5.0619607 , 5.069674  , 5.077373  , 5.0850606 ,
       5.0927343 , 5.1003957 , 5.108044  , 5.1156797 , 5.1233034 ,
       5.1309133 , 5.138512  , 5.146097  , 5.153671  , 5.1612315 ,
       5.1687803 , 5.1763163 , 5.1838408 , 5.191353  , 5.198853  ,
       5.2063403 , 5.213817  , 5.22128   , 5.2287326 , 5.2361727 ,
       5.243602  , 5.2510176 , 5.258424  , 5.2658167 , 5.2731986 ,
       5.2805686 , 5.2879276 , 5.295275  , 5.3030677 , 5.3103914 ,
       5.3177037 , 5.325005  , 5.3322945 , 5.339574  , 5.3468404 ,
       5.3540974 , 5.3613424 , 5.368577  , 5.3758006 , 5.3830132 ,
       5.3902154 , 5.397406  , 5.404587  , 5.4117556 , 5.418915  ,
       5.4260626 , 5.4332004 , 5.440327  , 5.4474435 , 5.4545503 ,
       5.461645  , 5.4687304 , 5.4758053 , 5.4828696 , 5.4899244 ,
       5.496968  , 5.504002  , 5.5110254 , 5.5180387 , 5.5250425 ,
       5.5320354 , 5.539019  , 5.5459924 , 5.5529556 , 5.55991   ,
       5.566853  , 5.573787  , 5.5807114 , 5.5876255 , 5.5945306 ],
      dtype=float32)
Coordinates:
  * x        (x) int64 5 6 7 8 9 10 11 12 13 ... 487 488 489 490 491 492 493 494
Attributes:
    wave_units:  microns
```

### Error traceback output

```
Traceback (most recent call last):
  File "/Users/acarter/Documents/TRANSITS/EUREKA/FIX_STAGE1/run_eureka.py", line 27, in <module>
    s3_spec, s3_meta = s3.reduce(eventlabel, ecf_path=ecf_path)#,
  File "/Users/acarter/Documents/TRANSITS/EUREKA/Eureka/src/eureka/S3_data_reduction/s3_reduce.py", line 446, in reduce
    plots_s3.lc_nodriftcorr(meta, spec.wave_1d, spec.optspec,
  File "/Users/acarter/Documents/TRANSITS/EUREKA/Eureka/src/eureka/S3_data_reduction/plots_s3.py", line 49, in lc_nodriftcorr
    plt.imshow(normspec, origin='lower', aspect='auto',
  File "/Users/acarter/anaconda3/envs/eureka16/lib/python3.9/site-packages/matplotlib/_api/deprecation.py", line 459, in wrapper
    return func(*args, **kwargs)
  File "/Users/acarter/anaconda3/envs/eureka16/lib/python3.9/site-packages/matplotlib/pyplot.py", line 2652, in imshow
    __ret = gca().imshow(
  File "/Users/acarter/anaconda3/envs/eureka16/lib/python3.9/site-packages/matplotlib/_api/deprecation.py", line 459, in wrapper
    return func(*args, **kwargs)
  File "/Users/acarter/anaconda3/envs/eureka16/lib/python3.9/site-packages/matplotlib/__init__.py", line 1412, in inner
    return func(ax, *map(sanitize_sequence, args), **kwargs)
  File "/Users/acarter/anaconda3/envs/eureka16/lib/python3.9/site-packages/matplotlib/axes/_axes.py", line 5491, in imshow
    im.set_extent(im.get_extent())
  File "/Users/acarter/anaconda3/envs/eureka16/lib/python3.9/site-packages/matplotlib/image.py", line 986, in set_extent
    self.axes.set_xlim((xmin, xmax), auto=None)
  File "/Users/acarter/anaconda3/envs/eureka16/lib/python3.9/site-packages/matplotlib/axes/_base.py", line 3697, in set_xlim
    left = self._validate_converted_limits(left, self.convert_xunits)
  File "/Users/acarter/anaconda3/envs/eureka16/lib/python3.9/site-packages/matplotlib/axes/_base.py", line 3614, in _validate_converted_limits
    raise ValueError("Axis limits cannot be NaN or Inf")
ValueError: Axis limits cannot be NaN or Inf
```

### What operating system are you using?

MacOS Catalina 10.15.7

### What version of Python are you running?

Python 3.9.7

### What Python packages do you have installed?

```
# packages in environment at /Users/acarter/anaconda3/envs/eureka16:
#
# Name                    Version                   Build  Channel
alabaster                 0.7.12                   pypi_0    pypi
appnope                   0.1.3                    pypi_0    pypi
asdf                      2.12.0                   pypi_0    pypi
asdf-astropy              0.2.1                    pypi_0    pypi
asdf-coordinates-schemas  0.1.0                    pypi_0    pypi
asdf-standard             1.0.2                    pypi_0    pypi
asdf-transform-schemas    0.2.2                    pypi_0    pypi
asdf-wcs-schemas          0.1.1                    pypi_0    pypi
asteval                   0.9.27                   pypi_0    pypi
astraeus                  0.2                      pypi_0    pypi
astropy                   5.1                      pypi_0    pypi
astropy-healpix           0.6                      pypi_0    pypi
astroquery                0.4.6                    pypi_0    pypi
astroscrappy              1.1.0                    pypi_0    pypi
asttokens                 2.0.5                    pypi_0    pypi
attrs                     21.4.0                   pypi_0    pypi
babel                     2.10.3                   pypi_0    pypi
backcall                  0.2.0                    pypi_0    pypi
batman-package            2.4.9                    pypi_0    pypi
bayesicfitting            3.0.1                    pypi_0    pypi
beautifulsoup4            4.11.1                   pypi_0    pypi
bokeh                     2.4.3                    pypi_0    pypi
bottleneck                1.3.5                    pypi_0    pypi
ca-certificates           2022.6.15            h033912b_0    conda-forge
ccdproc                   2.3.1                    pypi_0    pypi
celerite                  0.4.2                    pypi_0    pypi
certifi                   2022.6.15                pypi_0    pypi
cftime                    1.6.1                    pypi_0    pypi
charset-normalizer        2.1.0                    pypi_0    pypi
cloudpickle               2.1.0                    pypi_0    pypi
corner                    2.2.1                    pypi_0    pypi
crds                      11.16.5                  pypi_0    pypi
cycler                    0.11.0                   pypi_0    pypi
cython                    0.29.30                  pypi_0    pypi
dask                      2022.6.0                 pypi_0    pypi
decorator                 5.1.1                    pypi_0    pypi
docutils                  0.18.1                   pypi_0    pypi
drizzle                   1.13.4                   pypi_0    pypi
dynesty                   1.2.3                    pypi_0    pypi
emcee                     3.1.2                    pypi_0    pypi
eureka                    0.2.1                     dev_0    <develop>
executing                 0.8.3                    pypi_0    pypi
exotic-ld                 2.0.0                    pypi_0    pypi
filelock                  3.7.1                    pypi_0    pypi
fonttools                 4.34.4                   pypi_0    pypi
fsspec                    2022.5.0                 pypi_0    pypi
future                    0.18.2                   pypi_0    pypi
george                    0.4.0                    pypi_0    pypi
gwcs                      0.18.1                   pypi_0    pypi
h5netcdf                  1.0.1                    pypi_0    pypi
h5py                      3.1.0                    pypi_0    pypi
html5lib                  1.1                      pypi_0    pypi
idna                      3.3                      pypi_0    pypi
imageio                   2.19.3                   pypi_0    pypi
imagesize                 1.4.1                    pypi_0    pypi
importlib-metadata        4.12.0                   pypi_0    pypi
iniconfig                 1.1.1                    pypi_0    pypi
ipython                   8.4.0                    pypi_0    pypi
jedi                      0.18.1                   pypi_0    pypi
jinja2                    3.1.2                    pypi_0    pypi
jmespath                  1.0.1                    pypi_0    pypi
jsonschema                4.7.2                    pypi_0    pypi
jwst                      1.6.0                    pypi_0    pypi
keyring                   23.7.0                   pypi_0    pypi
kiwisolver                1.4.4                    pypi_0    pypi
libffi                    3.4.2                h0d85af4_5    conda-forge
libzlib                   1.2.12               hfe4f2af_2    conda-forge
lmfit                     1.0.3                    pypi_0    pypi
locket                    1.0.0                    pypi_0    pypi
lxml                      4.9.1                    pypi_0    pypi
markupsafe                2.1.1                    pypi_0    pypi
matplotlib                3.5.2                    pypi_0    pypi
matplotlib-inline         0.1.3                    pypi_0    pypi
ncurses                   6.3                  h96cf925_1    conda-forge
netcdf4                   1.6.0                    pypi_0    pypi
networkx                  2.8.4                    pypi_0    pypi
numpy                     1.23.1                   pypi_0    pypi
numpydoc                  1.4.0                    pypi_0    pypi
openssl                   3.0.5                hfe4f2af_0    conda-forge
packaging                 21.3                     pypi_0    pypi
pandas                    1.4.3                    pypi_0    pypi
parsley                   1.3                      pypi_0    pypi
parso                     0.8.3                    pypi_0    pypi
partd                     1.2.0                    pypi_0    pypi
pexpect                   4.8.0                    pypi_0    pypi
photutils                 1.5.0                    pypi_0    pypi
pickleshare               0.7.5                    pypi_0    pypi
pillow                    9.2.0                    pypi_0    pypi
pip                       22.1.2             pyhd8ed1ab_0    conda-forge
pluggy                    1.0.0                    pypi_0    pypi
poppy                     1.0.3                    pypi_0    pypi
prompt-toolkit            3.0.30                   pypi_0    pypi
psutil                    5.9.1                    pypi_0    pypi
ptyprocess                0.7.0                    pypi_0    pypi
pure-eval                 0.2.2                    pypi_0    pypi
py                        1.11.0                   pypi_0    pypi
pyerfa                    2.0.0.1                  pypi_0    pypi
pygments                  2.12.0                   pypi_0    pypi
pyparsing                 3.0.9                    pypi_0    pypi
pyrsistent                0.18.1                   pypi_0    pypi
pysynphot                 2.0.0                    pypi_0    pypi
pytest                    7.1.2                    pypi_0    pypi
python                    3.9.7           h38b4d05_3_cpython    conda-forge
python-dateutil           2.8.2                    pypi_0    pypi
python_abi                3.9                      2_cp39    conda-forge
pytz                      2022.1                   pypi_0    pypi
pyvo                      1.3                      pypi_0    pypi
pywavelets                1.3.0                    pypi_0    pypi
pyyaml                    6.0                      pypi_0    pypi
readline                  8.1.2                h3899abd_0    conda-forge
reproject                 0.8                      pypi_0    pypi
requests                  2.28.1                   pypi_0    pypi
scikit-image              0.19.3                   pypi_0    pypi
scipy                     1.8.1                    pypi_0    pypi
semantic-version          2.10.0                   pypi_0    pypi
setuptools                63.2.0           py39h6e9494a_0    conda-forge
six                       1.16.0                   pypi_0    pypi
snowballstemmer           2.2.0                    pypi_0    pypi
soupsieve                 2.3.2.post1              pypi_0    pypi
spherical-geometry        1.2.22                   pypi_0    pypi
sphinx                    5.0.2                    pypi_0    pypi
sphinxcontrib-applehelp   1.0.2                    pypi_0    pypi
sphinxcontrib-devhelp     1.0.2                    pypi_0    pypi
sphinxcontrib-htmlhelp    2.0.0                    pypi_0    pypi
sphinxcontrib-jsmath      1.0.1                    pypi_0    pypi
sphinxcontrib-qthelp      1.0.3                    pypi_0    pypi
sphinxcontrib-serializinghtml 1.1.5                    pypi_0    pypi
sqlite                    3.39.1               hd9f0692_0    conda-forge
stack-data                0.3.0                    pypi_0    pypi
stcal                     1.0.0                    pypi_0    pypi
stdatamodels              0.4.3                    pypi_0    pypi
stpipe                    0.4.1                    pypi_0    pypi
stsci-image               2.3.5                    pypi_0    pypi
stsci-imagestats          1.6.3                    pypi_0    pypi
stsci-stimage             0.2.5                    pypi_0    pypi
svo-filters               0.4.3                    pypi_0    pypi
tifffile                  2022.5.4                 pypi_0    pypi
tk                        8.6.12               h5dbffcc_0    conda-forge
tomli                     2.0.1                    pypi_0    pypi
toolz                     0.12.0                   pypi_0    pypi
tornado                   6.2                      pypi_0    pypi
tqdm                      4.64.0                   pypi_0    pypi
traitlets                 5.3.0                    pypi_0    pypi
tweakwcs                  0.7.4                    pypi_0    pypi
typing-extensions         4.3.0                    pypi_0    pypi
tzdata                    2022a                h191b570_0    conda-forge
uncertainties             3.1.7                    pypi_0    pypi
urllib3                   1.26.10                  pypi_0    pypi
wcwidth                   0.2.5                    pypi_0    pypi
webencodings              0.5.1                    pypi_0    pypi
wheel                     0.37.1             pyhd8ed1ab_0    conda-forge
wiimatch                  0.2.1                    pypi_0    pypi
xarray                    2022.3.0                 pypi_0    pypi
xz                        5.2.5                haf1e3a3_1    conda-forge
zipp                      3.8.1                    pypi_0    pypi
zlib                      1.2.12               hfe4f2af_2    conda-forge
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #412: [Enhancement]: 2D Lightcurve Minor Ticks
### Instrument

Light curve fitting (Stages 4-6)

### What is your suggestion?

It would be nice if we had minor ticks on the wavelength axis for the 2D lightcurve plots. It would make excluding bad parts of the wings of the spectrum slightly easier.

### Error traceback output

_No response_

### What operating system are you using?

_No response_

### What version of Python are you running?

_No response_

### What Python packages do you have installed?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #445: [Bug]: MIRI Wavelength Solution
### FAQ check

- [X] Yes, I checked the FAQ and my question has not been addressed.

### Instrument

MIRI (Stages 1-3)

### What happened?

Talking with Michael Zhang, I realized that our MIRI wavelength solution is likely a fair bit off. I don't know if it's a matter of an updated reference file, a bug in our original wavelength solution, or both. But at this point it looks like we may be off by ~4 pixels. We need to re-evaluate our wavelength solution with the most up-to-date wavelength solution reference files and ideally do things in a way that does not involve a hard-coded wavelength solution

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by taylorbell57**
This PR resolves #445, resolves #412, and resolves #399 (doesn't resolve NaN wavelengths but does avoid them causing the code to crash).

Most importantly, this PR fixes the incorrect wavelength solution for MIRI which was off by a few pixels. For now, this uses code that should eventually get merged into the `jwst` Stage 2 pipeline (after which point we shouldn't need to worry about fixing the completely zeroed out wavelength array). I also now track the original pixel indices for MIRI data since we mirror along the wavelength axis to have wavelength increase from left to right.

Secondly, this PR resolves an important bug with the x-axis on the 2D light curves. `plt.imshow` requires that the x-axis be linear, and at least for MIRI the dispersion is very non-constant and this lead to very wrong tick marks and very wrong vertical dashed lines in the S4 version of the figure. Thankfully [`plt.pcolormesh`](https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.pcolormesh.html#matplotlib-pyplot-pcolormesh) allows you to provide the x-axis and y-axis values and can be used to resolve this issue. pcolormesh uses pixels of non-uniform size and a linear x-axis, so for MIRI the 2D LC pixels are large at short wavelengths and are smaller at long wavelengths. As far as I can tell, it does not seem possible to keep the pixels constant size like imshow while having a non-linear x-axis when using pcolormesh. We might be able to manually hack non-linear x-ticks with an imshow figure, but I personally don't mind the behaviour of pcolormesh. While I was at it, I added minor tick marks as Yoni had requested.

Third, I also generalized Stage 6 so that you can make more plots than just transmission/emission spectra. There is now a separate file and plot made for every parameter you ask to plot, and I added some convenience features like allowing you to plot 1/r1 and 1/r4 to view the ramp timescales, rp^2 to view transit depths, and some default axis labels that are nicely formatted but user customizeable.

Finally, I also made some minor tweaks. I fixed the time units for the JWST instruments (BMJD_TDB not BJD_TDB). I also made some tweaks to the `clean_median_flux` function which were needed for my MIRI analysis - this will likely partially conflict with PR #450. I also made data['optmask'] a boolean array instead of an integer array for simplicity. I also tried to use .values whenever possible as not doing so can sometimes take a large amount of RAM or time. I also added `interpolation='nearest'` to relevant imshow figures where not using that left pixels blurred together.

### CODE REVIEW

**Comment by taylorbell57** [comment]
Alright, this PR should be good to go now - it runs successfully for me locally, and the only reason it fails on GitHub right now is that https://github.com/kevin218/Astraeus/pull/3 hasn't been merged so I can't add a new Astraeus version requirement yet to resolve the bug with that package

**Review by kevin218 — CHANGES_REQUESTED** [review]

File: src/eureka/S3_data_reduction/optspex.py [hunk 1]
```diff
@@ -481,7 +481,7 @@ def profile_gauss(subdata, mask, threshold=10, guess=None, isplots=0):
     return profile
 
 
-def clean_median_flux(data, meta, log, median_thresh=5):
```
[the remark this answers is not in the document]
Comment (taylorbell57) [reply src/eureka/S3_data_reduction/optspex.py:484]: Sure, I can merge main into my PR once yours is approved and merged.

File: src/eureka/S3_data_reduction/plots_s3.py [hunk 2]
```diff
@@ -39,21 +40,27 @@ def lc_nodriftcorr(meta, wave_1d, optspec, optmask=None):
               " Using 'y' by default.")
         meta.time_axis = 'y'
 
+    cmap = plt.cm.RdYlBu_r
     plt.figure(3101, figsize=(8, 8))
     plt.clf()
     if meta.time_axis == 'y':
-        plt.imshow(normspec, origin='lower', aspect='auto',
```
[the remark this answers is not in the document]
Comment (taylorbell57) [reply src/eureka/S3_data_reduction/plots_s3.py:45]: Sure, easy enough to do

File: src/eureka/S3_data_reduction/plots_s3.py [hunk 3]
```diff
@@ -94,7 +101,7 @@ def image_and_background(data, meta, log, m):
     median = np.ma.median(subbg)
     std = np.ma.std(subbg)
     # Set bad pixels to plot as black
-    cmap = mpl.cm.get_cmap("plasma").copy()
+    cmap = plt.cm.RdYlBu_r
```
[the remark this answers is not in the document]
Comment (taylorbell57) [reply src/eureka/S3_data_reduction/plots_s3.py:104]: I just did this as `plt.cm.RdYlBu_r` was used in the majority of cmaps and wanted to make things more uniform than using 3 different cmaps (without any reason for differences as far as I knew then). I don't know that it is overly important to have plasma and viridis to visually distinguish between the two different pixel position types when the axis labels should already do that. Do you still want to keep those two cmaps, or should I just switch them all to one cmap? And for those cmaps which were already `plt.cm.RdYlBu_r` (e.g. 2D LCs), do you have a preference for the cmap to use?

File: src/eureka/S3_data_reduction/plots_s3.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (taylorbell57) [reply src/eureka/S3_data_reduction/plots_s3.py:45]: One small issue with this is that all other 1-level figures will have to be incremented by one (3102-3106 already exist). Not a deal breaker, just wanted to make sure you know that's going to happen

[the remark this answers is not in the document]
Comment (taylorbell57) [reply src/eureka/S3_data_reduction/plots_s3.py:45]: Alright, here are the two versions as requested

fig3101-2D_LC
fig3102-2D_LC

File: src/eureka/S3_data_reduction/plots_s3.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (kevin218) [reply src/eureka/S3_data_reduction/plots_s3.py:134]: Let's return things to how they were for the viridis and plasma plots. If you want to pick another cmap for the 2D LC plot, that's fine.

File: src/eureka/S3_data_reduction/optspex.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (taylorbell57) [reply src/eureka/S3_data_reduction/optspex.py]: I just ended up reverting the changes here, so this PR is no longer in conflict with yours and they can be approved in whatever order.

**Comment by taylorbell57** [comment]
Alright, all the cmaps have been set back to what they were, and I reverted my changes to the clean_median_flux function so that this PR shouldn't conflict with yours anymore and we can merge whichever we want first. I think this one should be ready to merge now

File: src/eureka/S3_data_reduction/optspex.py [hunk 4]
```diff
@@ -535,7 +535,7 @@ def clean_median_flux(data, meta, log, m):
 
         # Flag outliers
         outliers = sigma_clip(residuals, sigma=meta.median_thresh, maxiters=5,
-                              axis=1, cenfunc='median')
+                              axis=1, cenfunc=np.ma.median, stdfunc=np.ma.std)
```
Comment (kevin218) [inline src/eureka/S3_data_reduction/optspex.py:538]: Looking at Figure 3401, this change doesn't have the desired effect for me (the entire region is yellow).  Please look at your output and verify the results.

File: src/eureka/S3_data_reduction/s3_reduce.py [hunk 5]
```diff
@@ -521,9 +520,10 @@ def reduce(eventlabel, ecf_path=None, s2_meta=None):
 
             # Compute MAD value
             if not meta.photometry:
-                meta.mad_s3 = util.get_mad(meta, log, spec.wave_1d,
-                                           spec.optspec,
-                                           optmask=spec.optmask)
+                meta.mad_s3 = util.get_mad(meta, log, spec.wave_1d.values,
+
```
Comment (kevin218) [inline src/eureka/S3_data_reduction/s3_reduce.py:524]: Looks like there's an extra blank line in the code

File: src/eureka/S3_data_reduction/plots_s3.py [hunk 6]
```diff
@@ -97,7 +133,7 @@ def image_and_background(data, meta, log, m):
     median = np.ma.median(subbg)
     std = np.ma.std(subbg)
     # Set bad pixels to plot as black
-    cmap = mpl.cm.get_cmap("plasma").copy()
+    cmap = plt.cm.plasma.copy()
```
Comment (kevin218) [inline src/eureka/S3_data_reduction/plots_s3.py:136]: I don't think we need `.copy()` with your new implementation (only with get_cmap).

File: src/eureka/S3_data_reduction/miri.py [hunk 7]
```diff
@@ -193,147 +205,23 @@ def wave_MIRI_hardcoded():
...
-                          [np.float64(np.nan)]*int(416-397.0-1))
-
-    return lam_x_full
+    if meta.firstFile:
+        log.writelog('  WARNING: Using the jwst package because the '
+                     'wavelengths are not currently in the .fits files '
+                     'themselves')
+
+    # Using the code from https://github.com/spacetelescope/jwst/pull/6964
+    # as of August 8th
```
Comment (kevin218) [inline src/eureka/S3_data_reduction/miri.py:217]: August 8th, 2022

File: src/eureka/S3_data_reduction/plots_s3.py (hunk 6 above)
Comment (taylorbell57) [reply src/eureka/S3_data_reduction/plots_s3.py:136]: `plt.cm.plasma` does return a static memory address (or so it seems), so if it get's called later I can't say with certainty that any changes to the cmap (e.g., bad colours/values) wouldn't end up affecting all the other places where that cmap was used

File: src/eureka/S3_data_reduction/optspex.py (hunk 4 above)
Comment (taylorbell57) [reply src/eureka/S3_data_reduction/optspex.py:538]: I had figured it'd be important to use those functions given we're using masked arrays, but it does seem to end up giving a significantly less smoothed median frame for me. I don't get the same behaviour as you (I'm using MIRI data), but I do get a worse median frame. However, the 2D LC with `cenfunc='median'` is far worse as it seems the sharp curvature in spectral response ends up getting partially clipped due to the smoothing (I used `window_len=11`). But either way, I've found that not smoothing works better for MIRI and if this change makes things worse for you, then I'll revert it

**Review by kevin218 — APPROVED** [review]

**Pull request merged**